src/packageurl/__init__.py

```diff
@@ -285,16 +285,21 @@ def __str__(self, *args, **kwargs):
     def __hash__(self):
         return hash(self.to_string())
 
-    def to_dict(self, encode=False):
+    def to_dict(self, encode=False, empty=None):
         """
-        Return an ordered dict of purl components as {key: value}. If `encode`
-        is True, then "qualifiers" are encoded as a normalized string.
-        Otherwise, qualifiers is a mapping.
+        Return an ordered dict of purl components as {key: value}.
+        If `encode` is True, then "qualifiers" are encoded as a normalized
+        string. Otherwise, qualifiers is a mapping.
+        You can provide a value for `empty` to be used in place of default None.
         """
         data = self._asdict()
         if encode:
             data['qualifiers'] = normalize_qualifiers(self.qualifiers,
                                                       encode=encode)
+
+        if empty is not None:
```

Comment:
Why using only there?

Reply:
Could you rephrase your question?

Reply:
Sorry for the poor wording ... I mean why would you only translate the empty values to `empty` if `empty` is not None?
Why not doing it all the time and therefore removing the test `if empty is not None:`?